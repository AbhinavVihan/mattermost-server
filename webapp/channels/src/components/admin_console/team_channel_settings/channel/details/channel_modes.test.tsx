// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {ChannelModes} from './channel_modes';
import {renderWithIntl} from 'tests/react_testing_utils';
import {Provider} from 'react-redux';
import store from 'stores/redux_store';

describe('admin_console/team_channel_settings/channel/ChannelModes', () => {
    test('should match snapshot', () => {
        const wrapper = renderWithIntl(
            <Provider store={store}>
                <ChannelModes
                    onToggle={jest.fn()}
                    isPublic={true}
                    isSynced={false}
                    isDefault={false}
                    isDisabled={false}
                    groupsSupported={true}
                />
            </Provider>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot - not licensed for Group', () => {
        const wrapper = renderWithIntl(
            <ChannelModes
                onToggle={jest.fn()}
                isPublic={true}
                isSynced={false}
                isDefault={false}
                isDisabled={false}
                groupsSupported={false}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
