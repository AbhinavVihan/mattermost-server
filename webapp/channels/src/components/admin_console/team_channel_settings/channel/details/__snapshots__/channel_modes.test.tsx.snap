// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin_console/team_channel_settings/channel/ChannelModes should match snapshot - not licensed for Group 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="AdminPanel clearfix "
        id="channel_manage"
      >
        <div
          class="header"
        >
          <div>
            <h3>
              Channel Management
            </h3>
            <div
              class="mt-2"
            >
              Choose between inviting members manually or syncing members automatically from groups.
            </div>
          </div>
        </div>
        <div
          class="group-teams-and-channels"
        >
          <div
            class="group-teams-and-channels--body"
          >
            <div>
              <div
                class="line-switch d-flex flex-sm-column flex-md-row align-items-sm-start align-items-center"
              >
                <label
                  class="line-switch__label"
                >
                  Public channel or private channel
                </label>
                <div
                  class="line-switch__toggle"
                  data-testid="allow-all-toggle"
                >
                  <button
                    aria-pressed="true"
                    class="btn btn-lg btn-toggle active"
                    data-testid="allow-all-toggle-button"
                    id="allow-all-toggle"
                    type="button"
                  >
                    <div
                      class="handle"
                    />
                    <div
                      class="bg-text on"
                    >
                      Public
                    </div>
                  </button>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col-sm-10"
                >
                  <div
                    class="help-text-small help-text-no-padding "
                  >
                    If \`public\` the channel is discoverable and any user can join, or if \`private\` invitations are required. Toggle to convert public channels to private. When Group Sync is enabled, private channels cannot be converted to public.
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="AdminPanel clearfix "
      id="channel_manage"
    >
      <div
        class="header"
      >
        <div>
          <h3>
            Channel Management
          </h3>
          <div
            class="mt-2"
          >
            Choose between inviting members manually or syncing members automatically from groups.
          </div>
        </div>
      </div>
      <div
        class="group-teams-and-channels"
      >
        <div
          class="group-teams-and-channels--body"
        >
          <div>
            <div
              class="line-switch d-flex flex-sm-column flex-md-row align-items-sm-start align-items-center"
            >
              <label
                class="line-switch__label"
              >
                Public channel or private channel
              </label>
              <div
                class="line-switch__toggle"
                data-testid="allow-all-toggle"
              >
                <button
                  aria-pressed="true"
                  class="btn btn-lg btn-toggle active"
                  data-testid="allow-all-toggle-button"
                  id="allow-all-toggle"
                  type="button"
                >
                  <div
                    class="handle"
                  />
                  <div
                    class="bg-text on"
                  >
                    Public
                  </div>
                </button>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-sm-10"
              >
                <div
                  class="help-text-small help-text-no-padding "
                >
                  If \`public\` the channel is discoverable and any user can join, or if \`private\` invitations are required. Toggle to convert public channels to private. When Group Sync is enabled, private channels cannot be converted to public.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`admin_console/team_channel_settings/channel/ChannelModes should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="AdminPanel clearfix "
        id="channel_manage"
      >
        <div
          class="header"
        >
          <div>
            <h3>
              Channel Management
            </h3>
            <div
              class="mt-2"
            >
              Choose between inviting members manually or syncing members automatically from groups.
            </div>
          </div>
        </div>
        <div
          class="group-teams-and-channels"
        >
          <div
            class="group-teams-and-channels--body"
          >
            <div>
              <div
                class="line-switch d-flex flex-sm-column flex-md-row align-items-sm-start align-items-center"
              >
                <label
                  class="line-switch__label"
                >
                  Sync Group Members
                </label>
                <div
                  class="line-switch__toggle"
                  data-testid="syncGroupSwitch"
                >
                  <button
                    aria-pressed="false"
                    class="btn btn-lg btn-toggle"
                    data-testid="syncGroupSwitch-button"
                    id="syncGroupSwitch"
                    type="button"
                  >
                    <div
                      class="handle"
                    />
                  </button>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col-sm-10"
                >
                  <div
                    class="help-text-small help-text-no-padding "
                  >
                    When enabled, adding and removing users from groups will add or remove them from this channel. The only way of inviting members to this channel is by adding the groups they belong to. 
                    <a
                      href="https://www.mattermost.com/pl/default-ldap-group-constrained-team-channel.html?utm_source=mattermost&utm_medium=in-product&utm_content=channel_modes&uid=&sid="
                      location="channel_modes"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Learn More
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="section-separator"
              >
                <hr
                  class="separator__hr"
                />
              </div>
            </div>
            <div>
              <div
                class="line-switch d-flex flex-sm-column flex-md-row align-items-sm-start align-items-center"
              >
                <label
                  class="line-switch__label"
                >
                  Public channel or private channel
                </label>
                <div
                  class="line-switch__toggle"
                  data-testid="allow-all-toggle"
                >
                  <button
                    aria-pressed="true"
                    class="btn btn-lg btn-toggle active"
                    data-testid="allow-all-toggle-button"
                    id="allow-all-toggle"
                    type="button"
                  >
                    <div
                      class="handle"
                    />
                    <div
                      class="bg-text on"
                    >
                      Public
                    </div>
                  </button>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col-sm-10"
                >
                  <div
                    class="help-text-small help-text-no-padding "
                  >
                    If \`public\` the channel is discoverable and any user can join, or if \`private\` invitations are required. Toggle to convert public channels to private. When Group Sync is enabled, private channels cannot be converted to public.
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="AdminPanel clearfix "
      id="channel_manage"
    >
      <div
        class="header"
      >
        <div>
          <h3>
            Channel Management
          </h3>
          <div
            class="mt-2"
          >
            Choose between inviting members manually or syncing members automatically from groups.
          </div>
        </div>
      </div>
      <div
        class="group-teams-and-channels"
      >
        <div
          class="group-teams-and-channels--body"
        >
          <div>
            <div
              class="line-switch d-flex flex-sm-column flex-md-row align-items-sm-start align-items-center"
            >
              <label
                class="line-switch__label"
              >
                Sync Group Members
              </label>
              <div
                class="line-switch__toggle"
                data-testid="syncGroupSwitch"
              >
                <button
                  aria-pressed="false"
                  class="btn btn-lg btn-toggle"
                  data-testid="syncGroupSwitch-button"
                  id="syncGroupSwitch"
                  type="button"
                >
                  <div
                    class="handle"
                  />
                </button>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-sm-10"
              >
                <div
                  class="help-text-small help-text-no-padding "
                >
                  When enabled, adding and removing users from groups will add or remove them from this channel. The only way of inviting members to this channel is by adding the groups they belong to. 
                  <a
                    href="https://www.mattermost.com/pl/default-ldap-group-constrained-team-channel.html?utm_source=mattermost&utm_medium=in-product&utm_content=channel_modes&uid=&sid="
                    location="channel_modes"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Learn More
                  </a>
                </div>
              </div>
            </div>
            <div
              class="section-separator"
            >
              <hr
                class="separator__hr"
              />
            </div>
          </div>
          <div>
            <div
              class="line-switch d-flex flex-sm-column flex-md-row align-items-sm-start align-items-center"
            >
              <label
                class="line-switch__label"
              >
                Public channel or private channel
              </label>
              <div
                class="line-switch__toggle"
                data-testid="allow-all-toggle"
              >
                <button
                  aria-pressed="true"
                  class="btn btn-lg btn-toggle active"
                  data-testid="allow-all-toggle-button"
                  id="allow-all-toggle"
                  type="button"
                >
                  <div
                    class="handle"
                  />
                  <div
                    class="bg-text on"
                  >
                    Public
                  </div>
                </button>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-sm-10"
              >
                <div
                  class="help-text-small help-text-no-padding "
                >
                  If \`public\` the channel is discoverable and any user can join, or if \`private\` invitations are required. Toggle to convert public channels to private. When Group Sync is enabled, private channels cannot be converted to public.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
